tools/docker/docker-entrypoint.sh

```diff
@@ -1,17 +1,26 @@
 #!/bin/bash
 
-CONFIG_FILE=assets/configuration/default.toml
+CONFIG_FILE_PATH=assets/configuration/
+CONFIG_FILE_NAME=default.toml
 
-# Modify the GN4 url and proxy path based on env variables (if defined)
-if [ ! -z "${GN4_API_URL}" ]
+# copy the conf file from /conf if present
+if [ -f "/conf/${CONFIG_FILE_NAME}" ]
 then
-  sed -i "s%geonetwork4_api_url = \".*\"%geonetwork4_api_url = \"${GN4_API_URL}\"%" /usr/share/nginx/html/${APP_NAME}/${CONFIG_FILE}
-  echo "[INFO] Replaced GN4 url in conf with: ${GN4_API_URL}"
-fi
-if [ ! -z "${PROXY_PATH}" ]
-then
-  sed -i "s%proxy_path = \".*\"%proxy_path = \"${PROXY_PATH}\"%" /usr/share/nginx/html/${APP_NAME}/${CONFIG_FILE}
-  echo "[INFO] Replaced proxy path in conf with: ${PROXY_PATH}"
+  cp /conf/${CONFIG_FILE_NAME} /usr/share/nginx/html/${APP_NAME}/${CONFIG_FILE_PATH}${CONFIG_FILE_NAME}
+  echo "[INFO] Copied custom configuration file located in /conf/${CONFIG_FILE_NAME}"
+else
+  # Modify the GN4 url and proxy path based on env variables (if defined)
+  if [ ! -z "${GN4_API_URL}" ]
+  then
+    sed -i "s%geonetwork4_api_url = \".*\"%geonetwork4_api_url = \"${GN4_API_URL}\"%" /usr/share/nginx/html/${APP_NAME}/${CONFIG_FILE_PATH}${CONFIG_FILE_NAME}
+    echo "[INFO] Replaced GN4 url in conf with: ${GN4_API_URL}"
+  fi
+  if [ ! -z "${PROXY_PATH}" ]
+  then
+    sed -i "s%proxy_path = \".*\"%proxy_path = \"${PROXY_PATH}\"%" /usr/share/nginx/html/${APP_NAME}/${CONFIG_FILE_PATH}${CONFIG_FILE_NAME}
+    echo "[INFO] Replaced proxy path in conf with: ${PROXY_PATH}"
+  fi
 fi
 
+
```

Comment:
I'm not sure to understand how we can override the config file, shall we edit the one from `assets/configuration/default.toml`, or from `conf/default.toml`, can we override the `CONFIG_FILE_NAME` ?

Reply:
to override the config file you need to essentially provide your own, custom file. For example, you can copy the `conf/default.toml` file in another directory, modify it, and provide this directory as a volume to the docker container.

Or you can just modify the `conf/default.toml` file in place and do:
```shell
$ docker run -v `pwd`/conf:/conf -p 8080:80 geonetwork-ui:datahub
```